src/blocks/remote-data-container/components/field-shortcode/FieldShortcodeButton.tsx

```diff
@@ -76,11 +79,20 @@ export function FieldShortcodeButton( props: WPFormatEditProps ) {
 	function onSelectField( data: FieldSelection, fieldValue: string ) {
 		updateOrInsertField( data, fieldValue );
 		onClose();
+		sendTracksEvent( 'remotedatablocks_field_shortcode', {
+			action: data.action,
+			data_source: getBlockDataSource( data.remoteData?.blockName ),
```

Comment:
seems like looking up the data source type should be delegated to `sendTracksEvent`

Reply:
We don't have the `data_source` prop on all events, so I didn't delegate it to `sendTracksEvent`.

Reply:
Can it just be undefined in those cases?

Reply:
It can be `undefined`, but we still need to pass `blockName`, so I think it's better to pass `data_source`, which is the value we ultimately send to `Tracks` anyway.

By the way, I've updated the types for these Tracks Events. (https://github.com/Automattic/remote-data-blocks/pull/185/commits/a14dfce570d4075348f37c5246b26fc89b93785c)